pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -564,23 +564,21 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 
 // dedupeDomains removes the duplicate domains from the passed in domains.
 func dedupeDomains(domains []string, vhdomains sets.String, expandedHosts []string, knownFQDNs sets.String) []string {
-	temp := domains[:0]
-	for _, d := range domains {
+	return slices.FilterInPlace(domains, func(d string) bool {
 		if vhdomains.Contains(strings.ToLower(d)) {
-			continue
+			return false
 		}
 		// Check if the domain is an "expanded" host, and its also a known FQDN
 		// This prevents a case where a domain like "foo.com.cluster.local" gets expanded to "foo.com", overwriting
 		// the real "foo.com"
 		// This works by providing a list of domains that were added as expanding the DNS domain as part of expandedHosts,
 		// and a list of known unexpanded FQDNs to compare against
 		if slices.Contains(expandedHosts, d) && knownFQDNs.Contains(d) { // O(n) search, but n is at most 10
-			continue
+			return false
 		}
-		temp = append(temp, d)
 		vhdomains.Insert(strings.ToLower(d))
```

Comment:
a bit worried about this TBH. We shouldn't be mutating state in something like `FilterInPlace`. It works today but there is no guarantees on how `FilterInPlace` is implemented

Reply:
After thinking for a few minutes, I didn't realize what you were worried about. However, to avoid possible risks, I revert this change. Thanks :)